Submission of Checkpoint 1B

sslgeorge opened the pull request:
Completed project for Checkpoint 1B, implemented the following:
1.  Call to Github API for particular `user`
2.  Classification of `user` based on the number of repositories they own
    Categories include `Junior (5-10)`, `Associate(11-20)`, `Senior(21 and above)` and `Baby (Less than     5)` for developers.
3.  Test of each class created

Commit:
(Feature) Created a github client object

To enable access to users github account, a class was created to abstract the
functinality of connecting to users account and getting their repositories

Commit:
(Chore) Added integration for StyleCI, TravisCI and CirleCI

To ensure comformation with standards, support for code style, and unit test
check is added to the project

Commit:
(Feature) Added class to get Evangelist Information

This class will enable users check their evangelist status, functionality
to get all user repos added.

Commit:
Applied fixes from StyleCI

Commit:
Merge pull request #1 from andela-gjames/analysis-XVmBZ8

(Chore) Applied fixes from StyleCI

Commit:
(Feature)(Fix) Added functionality to get number of repositories

Added new features/functions to:
1.  Set the username for the github user
2. Get Repositories of the user
3. Get the number of repositories each individual has

getRepos() function refactored to getRepositoris()

Commit:
(Chore)Merge branch 'develop' of https://github.com/andela-gjames/Evangelizers into develop

Fixed Conflicts Created by StyleCI:
	src/GithubClient.php

Commit:
(Feature) Added dynamic property call

Added the following feature:
1.	Dynamic Method Call: made use of __call magic method to enable users make call to any available
	property from the github response
2. 	Added function getRepositoriesName(): This method is added to make call to getting names of
	repositories feel natural.

Commit:
(Feature) Added ability to get user status

Features added to EvangelistClass
1.	getStatus function: This function returns a text relating to the level of the owner of the github repo
2.	protected function defineValue: this function is used to decide the level of the user
3.	protected function getStatusQuote: This function determines the value saved in the protected property status and
	based on the result serves the appropriate text for the user based on the number of repositories owned by that user.

Commit:
(Chore) Added comments and formating to files

Commit:
(Chore) Made to implement the APIClient

Since an interface has been created to abstract the concept of the APIClient
it now makes sent for the GithubClient class to implement the interface

Commit:
(Chore) Comment the class for easy reading of code

Commit:
(Feature) Added an interface to abstract API calls

In order to accomodate usage of more than one API, the concept of
the API is abstracted into a new method

Commit:
(Feature) Added a new custome Exception

New exception has been created, to ensure that understable exception calls are made

Commit:
(Fix) Catch errors thrown from exception

Ensure that all thrown excptions are caught and properly handled

Commit:
(Fix) Refactored code to implement new custome Exception class

Commit:
(Chore) Added phpunit.xml to bootstrap the phpunit tests

Commit:
(Feature) Added test scripts for classes created

In order to test all programs and ensure that they perform as expected
tests were written to assert truthy.
1:	Test class was created for GithubClient class
2:	Test class was created for EvangelistStatius class
3.	Mock File was created for the GithubClient class so simulate
	access to online github api

Commit:
(Chore) Added README.md file

Commit:
(Chore) Updated travis.yml and cirlceci.yml

Commit:
Applied fixes from StyleCI

Commit:
Merge pull request #2 from andela-gjames/analysis-XpARRq

(Chore) Applied fixes from StyleCI

Commit:
(Chore) Update README.md

Commit:
(Chore) Update README.md

Commit:
(Chore) Updated CirclCi.yml

Commit:
(Chore) Update Composer.json

Commit:
Updated cycleCI.json

Commit:
(Chore) Update Composer.json

Commit:
(Chore) Update travis.yml

Commit:
(Chore) Added composer.lock to version control

composer.lock file was initially blocked by gitignore, but from research, I find
that the composer.lock file is necessary for continous integration

Commit:
(Chore) Changed PHP requirements from composer

Commit:
(Chore) Fixing TravisCI and CircleCi

Commit:
(Chore) Added CycleCI badge for passing develop branch

Commit:
(Chore) Integrated Coverall to project

To enable Coverall run:
1. added .coveralls.yml: set src directory to project root
2. Updated .gitgnore: added composer.lock to .gitignore
3. updated .travis.yml: added the coveralls key, override default phpunit test to use phpunit.xml for bootstraping etc
4. README.md: updated to feature coveralls badge
5. composer.json: added the phpcoveralls package to project

Commit:
(chore) Added Travis pass badge to test Coverall Integration

Commit:
(Chore) Added logging path for coveralls in phpunit.xml

Commit:
(Chore) Regenerated key for coveralls added

Commit:
(Chore) Removed composer.lock

Commit:
Update README.md

Commit:
(Chore) Added more tests and comments

The following were done in this commit:
1.	I added more test, I broke down the test for the EvangelistStatus to be more specific
2.  	Added more data mocks to make test not dependent on 3rd party service
3.	Some exceptions will never be thrown, so I ensured to fix such issues
4. 	Added comments for documenting functions.

Commit:
(Chore) Merged, and fixed failing test

Pulled changed files from Github, and fixed failing test for Junior developer

Commit:
(Chore) Merge remote branch 'develop' into develop

Commit:
Applied fixes from StyleCI

Commit:
(chore) Merge branch remote 'develop' into develop

Commit:
Merge pull request #3 from andela-gjames/analysis-qvjgEq

(Chore) Applied fixes from StyleCI

Commit:
(Chore) Exposed more test areas

Edited test for GithubClient to ensure more test areaas are covered

andela-anandaa commented:
#### 1. Comments / README
- Well commented code
- The README is good; also good use of the StyleCI, CircleCI and Coveralls badges

#### 2. Functionality
- Works as expected

#### 3. Code Quality
- Good design patterns: interface, magic method, custom exceptions, mocks
- Code is compliant to PSR 2 standards
- Very good understanding of OOP
- Simplify the `if-else` block for _Evangelist Status_  (nitpick)

#### 4. Testing
- Good integration with CircleCI + Coveralls integration
- You can place the json files in a directory `fixtures`
- Test-coverage is excellent, 91%!

#### 5. Commit messages
- Atomic, consistently formatted and well structured commit messages
  General
- Great walk-through the code, and explaining the design choices.

Commit:
(Chore) Refactored code

The following were carried out:
1.	Moved all JSON faker files to /tests/Mocks/Fixtures
2.	Refactored EvangelistStatusTest.php and GithubCLientTest.php to feature new dir structure
3.	Refactored if-else statement in src/EvangelistStatus.php to be clearer and more readable
